cmake/herbert_Version.cmake

```diff
@@ -1,38 +1,45 @@
+#[=======================================================================[.rst:
+herbert_Version
+-----------------
+
+Build a version release string:
+  <version>[-<date>]-<target>-<matlab>[-<sha>]
+
+ Optional elements are included based on the value of Herbert_RELEASE_TYPE
+ - Date included for "nightly" builds
+ - SHA included for non-"release" builds (i.e. "nightly" or "pull-request"
+
+Variables required by the module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Herbert_RELEASE_TYPE
+Release type: "nightly", "release" or "pull-request"
+
+``Matlab_VERSION
+This is provided by the `herbert_FindMatlab` module which must be loaded first
+
+Variables defined by the module
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Herbert_FULL_VERSION``
+formatted version string
+
+#]=======================================================================]
+
 set(Herbert_FULL_VERSION "${PROJECT_VERSION}")
 
 if(Herbert_RELEASE_TYPE STREQUAL "nightly")
     string(TIMESTAMP _date "%Y%m%d")
     set(Herbert_FULL_VERSION "${Herbert_FULL_VERSION}-${_date}")
 endif()
 
-macro(get_release_value)
-    set(_release_key ${ARGV0})
-    execute_process(
-        COMMAND cat "/etc/os-release"
-        COMMAND grep "-e" "^${_release_key}="
-        RESULT_VARIABLE _res
-        OUTPUT_VARIABLE ${ARGV1}
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    string(REGEX REPLACE
-        "${_release_key}=\\\"?\(.+\)\\\"?"
-        "\\1"
-        ${ARGV1}
-        ${${ARGV1}}
-    )
-    string(REPLACE " " "." ${ARGV1} ${${ARGV1}})
-    string(REPLACE "\"" "" ${ARGV1} ${${ARGV1}})
-endmacro()
-
 if(UNIX)
-    # Get distribution name from /etc/os-release
-    get_release_value("ID" _id)
-    get_release_value("VERSION_ID" _version_id)
-    set(Herbert_PLATFORM "${_id}.${_version_id}")
+    set(Herbert_PLATFORM "linux")
```

Comment:
It might make sense to convert this block into 
```cmake
if(APPLE)
  ...
elseif(UNIX)
  ...
elseif(WIN32)
  ...
endif()

```

Note I think `APPLE` needs to come before `if(UNIX)`, since `if(UNIX)` can evaluate to true on some apple systems

Reply:
The APPLE branch will be added as part of https://github.com/pace-neutrons/Horace/issues/47